src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.ReadCore.cs

```diff
@@ -1,18 +1,21 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Diagnostics;
+
 namespace System.Text.Json.Serialization
 {
     public partial class JsonConverter<T>
     {
-        internal T? ReadCore(
+        internal bool ReadCore(
```

Comment:
This changes the shape of the method a `Try-` style method so that the converter has control over when to stop the async deserialization. This was originally controlled by the async loop that would keep calling into the converter until the underlying stream was read to its end.